shanghai/core_plugins/channel.py

```diff
@@ -308,3 +381,36 @@ def chan_eq(ctx: NetworkContext, chan1: str, chan2: str):
     ctx.message_event('KICK')(on_kick)
     ctx.message_event('NICK')(on_nick)
     ctx.message_event('QUIT')(on_quit)
+
+    channel_contexts = {}
+    channel_event_dispatcher = ChannelEventDispatcher()
+    ctx.add_attribute('channel_event', channel_event_dispatcher.decorator)
+
+    @ctx.add_method
+    def get_channel_context(n_ctx: NetworkContext, channel: str):
+        lchannel = n_ctx.chan_lower(channel)
+        return channel_contexts[lchannel]
+
+    @ctx.message_event('PRIVMSG')
+    @ctx.message_event('NOTICE')
```

Comment:
By the way, using multiple decorators for the same function is intended to work but not well-tested at the moment. Specifically with regard to unregistering. Feel free to use it, however, since it's on my radar.